src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -245,9 +252,44 @@ public void setWebhookRegistered(boolean webhookRegistered) {
     protected void retrieve(@CheckForNull SCMSourceCriteria criteria, SCMHeadObserver observer,
                             @CheckForNull SCMHeadEvent<?> event,
                             TaskListener listener) throws IOException, InterruptedException {
+        if (event == null) {
+            //on manual scans & creation of jenkins jobs, we call out to bbs and fetch a list of open prs and
+            // sync up our local pullRequestStore
+            DescriptorImpl descriptor = (DescriptorImpl) getDescriptor();
+            List<BitbucketPullRequest> bbsPullRequests = fetchOpenPullRequestsFromBbsInstance(descriptor);
+            String serverId = getServerId();
+            if (serverId != null) {
+                descriptor.getPullRequestStore().refreshStore(getProjectKey(), getRepositorySlug(), serverId, bbsPullRequests);
+            }
+        }
         gitSCMSource.accessibleRetrieve(criteria, observer, event, listener);
     }
 
+    private List<BitbucketPullRequest> fetchOpenPullRequestsFromBbsInstance(DescriptorImpl descriptor) {
+        BitbucketServerConfiguration bitbucketServerConfiguration = descriptor.getConfiguration(getServerId())
+                .orElseThrow(() -> new BitbucketClientException(
```

Comment:
I don't think we can do this. I think we need to either catch this exception at the caller and proceed to do a scan anyway, or not throw it here.